### ISSUE TITLE
Deferred Resolvers

### ISSUE THREAD

**Issue by asheliahut** [digiaonline/graphql-php#287]
### What did you do?

I created a resolver that resolves to another type.

### What did you expect to happen?

I expected to be able to grab all sub types in a single database query.

### What actually happened?

I was only able to resolve to a single type each time for all the sub types which led to 100s of database queries.

### What version of this project are you using?
master branch.

Please include code that reproduces the issue. The best reproductions are self-contained scripts with minimal dependencies.

Code is not included as this is a new feature.

**Comment by hungneox** [comment]
@asheliahut Hi, thanks for reaching out! It would be great if you can provide a piece of example code 😃

> Please include code that reproduces the issue. The best reproductions are self-contained scripts with minimal dependencies.

As far as I understand. Do you mean this kind of query below doesn't resolve all sub types?:

```
{
   mainType(...) {
        id
        subType1 {
            id
        }
        subType2 {
            id
        } 
    }
}
```

**Comment by hungneox** [comment]
@asheliahut Ok, thanks. Now I understand the issue.

**Comment by hungneox** [comment]
@asheliahut Hello again, is this enough to close this issue? https://github.com/digiaonline/graphql-php/pull/291 
The feature is already implemented, I have just added a test.

**Comment by asheliahut** [comment]
@hungneox it is in fact solved by that. It was my thought from my second comment.

**Comment by crisu83** [comment]
@asheliahut why did you re-open this, because of missing docs?

**Comment by asheliahut** [comment]
@crisu83 yes I reopened it because I personally closed it by accident. I typically when tracking issues link the PR that fixes the issue. So I reopened it until the PR was closed.

**Comment by crisu83** [comment]
@asheliahut alright, we'll keep it open until the PR is merged then. 👍
